### PR TITLE
Modifications to Swagger handler

### PR DESCRIPTION
**Pull request by h-parekh**
* Adding support for private git repositories
* Parameterizing swagger definition URL based on repository URL from a config file

### CODE REVIEW

File: spec/spec_support/swagger_handler.rb [hunk 1]
```diff
@@ -20,20 +22,27 @@ def operations(&block)
 
     private
 
-    def project_user_name
-      'ndlib'
-    end
-
-    def project_repository_name
-      'test_api'
+    def   read_repo_config
```
Comment (jeremyf) [inline spec/spec_support/swagger_handler.rb]: one space between def and the method name.

Comment (jeremyf) [inline spec/spec_support/swagger_handler.rb]: I recommend that you set all instance variables as part of the initialize process. Then, you can call the other methods that use the instance variables at a later point. This helps clean up and separate concerns of the code.

File: spec/viceroy/integration/int_viceroy_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+require 'viceroy/viceroy_spec_helper'
+
+describe 'API tests using swagger definition' do
+  SwaggerHandler.operations(for_file_path: __FILE__).each do |operation|
```
Comment (jeremyf) [inline spec/viceroy/integration/int_viceroy_spec.rb]: I'm fine with this, but now that we have two use cases of Swagger operations, we should look to remove the duplication of knowledge. It gets a bit tricky, but I can help on that behavior.

File: spec/spec_support/swagger_handler.rb (hunk 1 above)
Comment (h-parekh) [inline spec/spec_support/swagger_handler.rb]: I moved 'read_repo_config' and 'read_git_access_token' under initialize.

**Pull request merged**